server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,278 @@
+import { Message, GuildTextableChannel, TextableChannel, User, EmbedOptions, Client } from 'eris'
+import { Command, DB } from '../imports/types'
+import fs from 'fs'
+import { promisify } from 'util'
+import { getInsult } from '../imports/tools'
+
+async function parseTriviaList (fileName: string) {
+  const readFile = promisify(fs.readFile)
+  const data = await readFile(`./triviaLists/${fileName}.txt`, 'utf8')
+  const triviaList = data.split('\n')
+  return triviaList
+    .map(el => el.replace('\n', '').split('`'))
+    .filter(el => el.length >= 2 && el[0])
+    .map(el => ({
+      question: el[0],
+      answers: el.slice(1).map(ans => ans.trim())
+    }))
+}
+
+export class TriviaSession {
+  revealMessages = ['I know this: ', 'Easy: ', 'Of course, it\'s: ']
+  failMessages = ['You suck at this', 'That was trivial, really', 'Moving on...']
+  settings: { maxScore: number, timeout: number, delay: number, botPlays: boolean, revealAnswer: boolean }
+  currentLine: {question: string, answers: string[]}
+  channel: TextableChannel
+  author: User
+  message: Message
+  questionList: {question: string, answers: string[]}[]
+  scores: { [char: string]: number } = {}
+  status = ''
```

Comment:
Maybe use an enum, or type this like so?
```suggestion
  status: 'correct answer' | 'stop' | 'new question' | 'waiting for answer' = ''
```
Enum approach is better though.

Reply:
I haven't looked at the code closely but do we need both `correct answer` and `new question`? Why not just have `processing` instead, or even better, `waiting: boolean | null` where false is `stopped`, null is `processing` and true is `waiting for answers`.

Reply:
A few conclusions:
- `new question` is completely unnecessary. Why? Because whenever it's set to `this.status = 'new question'`, `newQuestion()` is called immediately afterwards which resets `status` to `waiting for answer`.
- "Oh, but there's a 1 second wait before it's called in which `checkAnswer` might get called?" `checkAnswer` doesn't rely on the value of `this.status`, it uses `currentLine` instead by checking if it's null or not.
- So, `new question` should be dropped.
- On top of that, `correct answer` should be dropped too, instead just check if `currentLine` is null or not. Why? `currentLine` is always non-null when `status` is `waiting for answer` and null when `correct answer`.
- This means that you drop `correct answer` by replacing its usages i.e. `status === 'correct answer'` with `currentLine === null` and `status !== 'correct answer'` with `currentLine !== null`.
- And drop `waiting for answer` by replacing its usages i.e. `status === 'waiting for answer'` with `currentLine !== null` and `status !== 'waiting for answer'` with `currentLine === null`.
- This leaves us with `'stop'`. `stop` can be checked like so: `const stopped = this.tempDB.trivia[this.channel.id] !== this`. But for concision you can just convert `status` to `stopped: boolean` which is more appropriate.